ash-linux/el7/STIGbyID/cat2/RHEL-07-020320.sls

```
@@ -0,0 +1,102 @@
+                         'jfs',
+                         'btrfs'
+                        ] %}
+{%- set mountData = salt.mount.fstab() %}
+{%- set mounts = mountData.keys() %}
+
+script_{{ stig_id }}-describe:
+  cmd.script:
+    - source: salt://{{ helperLoc }}/{{ stig_id }}.sh
+    - cwd: /root
+
+{%- if stig_id in skipIt %}
+notify_{{ stig_id }}-skipSet:
+  cmd.run:
+    - name: 'printf "\nchanged=no comment=''Handler for {{ stig_id }} has been selected for skip.''\n"'
+    - stateful: True
+    - cwd: /root
+{%- else %}
+# Find files with no valid owner..
+  {%- for mount in mounts %}
+    {%- set mountType = mountData[mount]['fstype'] %}
+    {%- if mountData[mount]['fstype'] in localFstypes %}
+      {%- set foundString = salt['cmd.shell']('find ' + mount + ' -xdev -type f -nouser') %}
+      {%- if foundString %}
+        {%- set foundList = foundString.split('\n') %}
+        {%- do nouserFiles.extend(foundList) %}
+      {%- endif %}
+    {%- endif %}
+  {%- endfor %}
+
+# Find directories with no valid owner..
+  {%- for mount in mounts %}
+    {%- set mountType = mountData[mount]['fstype'] %}
+    {%- if mountData[mount]['fstype'] in localFstypes %}
+      {%- set foundString = salt['cmd.shell']('find ' + mount + ' -xdev -type d -nouser') %}
+      {%- if foundString %}
+        {%- set foundList = foundString.split('\n') %}
+        {%- do nouserDirs.extend(foundList) %}
+      {%- endif %}
+    {%- endif %}
+  {%- endfor %}
+
+# Take ownership of files
+  {%- if nouserFiles|length %}
+    {%- for file in nouserFiles %}
+      {%- if file %}
+file_{{ stig_id }}-{{ file }}:
+  file.managed:
+    - name: '{{ file }}'
+    - user: 'root'
+      {%- endif %}
+    {%- endfor %}
+  {%- else %}
+file_{{ stig_id }}-noneFound:
+  cmd.run:
+    - name: 'printf "\nchanged=no comment=''Found no files with missing/undefined users.''\n"'
+    - stateful: True
+    - cwd: /root
+  {%- endif %}
+
+# Take ownership of directories
+  {%- if nouserDirs|length %}
```

The length filter isn't needed

```suggestion
  {%- if nouserDirs %}
```

---

Leftover from before I determined that my if/else was failing due to my list being a list of nulls rather than a null list